src/System.Web.OData/OData/Routing/DefaultODataPathHandler.cs

```diff
@@ -112,12 +112,10 @@ private ODataPath Parse(string serviceRoot, string odataPath, IServiceProvider r
             {
                 Contract.Assert(serviceRoot != null);
 
-                serviceRootUri = new Uri(
-                    serviceRoot.EndsWith("/", StringComparison.Ordinal)
-                        ? serviceRoot
-                        : serviceRoot + "/");
+                string serviceRootStr = serviceRoot.EndsWith("/", StringComparison.Ordinal) ? serviceRoot : serviceRoot + "/";
+                serviceRootUri = new Uri(serviceRootStr);
```

Comment:
serviceRootUri is used below.
This is the original code.
uriParser = new ODataUriParser(model, serviceRootUri, fullUri, requestContainer);